backend/src/index.ts

```diff
@@ -0,0 +1,14 @@
+import express from 'express';
+import cors from 'cors';
+
+const app = express();
+app.use(express.json());
+app.use(cors());
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/1#issuecomment-2348763797>#3e1861</a><span> - 09/13/2024, 11:56 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> CORS middleware is used without any configuration, potentially allowing all origins and creating security vulnerabilities.
</div>
<div id="fix">
<b>Fix:</b> Configure CORS with specific allowed origins, methods, and headers to restrict access to trusted domains only.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,8 +1,15 @@
  import express from 'express';
  import cors from 'cors';
 
  const app = express();
  app.use(express.json());
 -app.use(cors());
 +app.use(cors({
 +  origin: process.env.ALLOWED_ORIGINS?.split(',') || 'http://localhost:3000',
 +  methods: ['GET', 'POST', 'PUT', 'DELETE'],
 +  allowedHeaders: ['Content-Type', 'Authorization'],
 +  credentials: true,
 +  optionsSuccessStatus: 204,
 +}));
 
  app.get('/', (req, res) => {
    res.send('Hello World!');
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged